tree.go

```diff
@@ -360,6 +360,15 @@ func (n *InternalNode) Delete(key []byte) error {
 		return nil
 	default:
 		child.Delete(key)
+		// Prune child if necessary
+		nonEmpty := child.(*InternalNode).nonEmptyIndices()
```

Comment:
I'm not a big fan of allocating memory just to get the size, here. Wouldn't it be possible to have one loop counting non-`nil` children, and bails if the count is greater than 2?